src/main/java/backend/UpdateController.java

```diff
@@ -112,11 +113,12 @@ private void openRepositoriesInFilters(List<FilterExpression> filterExprs) {
                         .sorted(determineComparator(filterExprNoAlias, hasUpdatedQualifier))
                         .collect(Collectors.toList());
 
-                filteredAndSorted.put(filterExpr, filteredAndSortedIssues);
+                filteredSortedAndCounted.put(filterExpr,
```

Comment:
Instead of doing this, you can add `.limit(n)` after `.sorted(...)`. That will take care of truncating the stream so only n or fewer elements remain after the entire thing is collected.